### ISSUE TITLE
Declare `@ccall` only when asked by the user

### ISSUE THREAD

**Issue by czgdp1807** [lcompilers/lpython#1094]
So as of now any function declared with `@ccall` is declared always in the generated code. However if someone has already declared their function in some other `.h` file then function decorated `@ccall` shouldn't be re-declared in the generated code. So we have to make the declaration of `@ccall` functions optional. Some options,

1. `ccall(static=true)`.
2. `ccall(declare=false)`.

**Comment by czgdp1807** [comment]
@certik @rebcabin

**Comment by czgdp1807** [comment]
I see. If there are large number of such functions then adding flags for each function in the `.h` would be difficult for the user. Also if the `.h` file is uneditable due to some reasons then `ifndef` approach may not work.

**Comment by czgdp1807** [comment]
So after discussion with @rebcabin we figured that all the `ccall` decorated functions can go into `#ifndef IGNORE_REDECLARATIONS … #endif` block. If the user defines  in their headers then `IGNORE_REDECLARATIONS` or in a build file then all the redeclarations in LPython’s generated C code will be ignored.

**Comment by czgdp1807** [comment]
Well seems like C compiler doesn't raise an error when we declare a function which is already declared in some header we have included. 

**Example 1** 

See https://github.com/czgdp1807/lpython/commit/4fa09ae9be3e34ece9a9e6fdc8c01c1a6525c8ac

Instructions (assuming you are in project root already)

* `cd integration_tests`
* `mkdir b3`
* `cmake -DKIND=c ..`
* `make bindc_05` (succeeds)
* `./bindc_05`

**Example 2**

```c
#include <math.h>
#include <stdio.h>

// redeclaring pow function from math.h
double pow(double, double);

int main() {
    double a = 2.0, b = 2.0;
    double ans = pow(a, b);
    printf("%lf\n", ans);
    return 0;
}
```

Succeeds and gives `4.0000` as output. 

So I am unable to re-produce an example which causes the problem as speculated in the issue description. @ronnuriel If you have any re-producer of the problem then please feel free to share it.

**Comment by ronnuriel** [comment]
@czgdp1807 the compile did raise an error but I will check that again and get back to you

**Comment by czgdp1807** [comment]
Sure. Send me an example of the problem (with the code, the instructions to run the code and the error you get). In the second example (where I re-declare `pow`) I simply used my compiler to generate the executable (no build systems of any kind). AFAICT, C compiler ignores the forward declarations if you are already providing it in a header file.

**Comment by Smit-create** [comment]
> Well seems like C compiler doesn't raise an error when we declare a function which is already declared in some header we have included.

I also didn't get any error/warning for the same example.

**Comment by ronnuriel** [comment]
I've checked that issue and I had the "redundant-decls" flag set on, I removed it and it did solve the error, but now we are facing another issue:
In lpython we cant use `enum mmb_vr16` as a variable type so we using `i16` so its basically the same but different and that causes the next compilation error:
while the function declared with enum mmb_vr16 and we declared it with int16_t
**EXAMPLE:**

**Header file deceleration:** 
`void csrc_cpy_16_vmrk_t0(enum mmb_vr16 dst, enum mmb_vr16 src, enum mmb_vr16 vmrk, u16 mrk);`

**Lpython deceleration:**
`void csrc_cpy_16_vmrk_t0(int16_t dst, int16_t src, int16_t vmrk, int16_t mrk);`

This issue cant be solved this with:

1. ccall(static=true).
2. ccall(declare=false).
